src/Modules/ModuleTypes.jl

```diff
@@ -0,0 +1,563 @@
+export ModuleFP, ModuleFPElem, ModuleFPHom, ModuleMap, FreeMod,
+       FreeModElem, SubQuo, SubQuoElem, FreeModuleHom, SubQuoHom,
+       FreeMod_dec, FreeModElem_dec, FreeModuleHom_dec
+
+@doc Markdown.doc"""
+    ModuleFP{T}
+
+The abstract supertype of all modules. Here, all modules are finitely presented.
```

Comment:
Isn't an oxymoron? So perhaps:
```suggestion
The abstract supertype of all finitely presented modules.
```